.github/workflows/changelog-check.yml

```diff
@@ -0,0 +1,107 @@
+    pull_request:
+        branches: [develop]
+    # Pushes to develop and main/master require the changelog to be updated
+    push:
+        branches:
+            - develop
+            - main
+            - master
+
+jobs:
+    check-changelog:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+              with:
+                  fetch-depth: 0  # Fetch full history
+
+            - name: Check if CHANGELOG.md was modified or bypassed
+              id: changelog-check
+              run: |
+                  # For PR events, compare against the target branch
+                  if [ "${{ github.event_name }}" == "pull_request" ]; then
+                      BASE_SHA="${{ github.event.pull_request.base.sha }}"
+                      HEAD_SHA="${{ github.event.pull_request.head.sha }}"
+                      echo "Checking PR commits from $BASE_SHA to $HEAD_SHA"
+                  else
+                      # For push events, compare with the previous commit
+                      BASE_SHA="${{ github.event.before }}"
+                      HEAD_SHA="${{ github.sha }}"
+                      echo "Checking push commits from $BASE_SHA to $HEAD_SHA"
+                  fi
+
+                  # Get all commit messages in the range
+                  echo "Getting all commit messages in range..."
+                  if [ "$BASE_SHA" == "0000000000000000000000000000000000000000" ]; then
+                      # New branch, just check the latest commit
+                      COMMIT_MESSAGES=$(git log -1 --pretty=format:"%B" "$HEAD_SHA")
+                  else
+                      COMMIT_MESSAGES=$(git log --pretty=format:"%B" "$BASE_SHA".."$HEAD_SHA")
+                  fi
+
+                  # Check if CHANGELOG.md was modified
+                  if git diff --name-only "$BASE_SHA" "$HEAD_SHA" | grep -q "^CHANGELOG.md$"; then
+                      echo "✅ CHANGELOG.md has been modified - check passed"
+                      echo "bypassed=false" >> $GITHUB_OUTPUT
+                  # Check for bypass flag in ANY commit message
+                  elif echo "$COMMIT_MESSAGES" | grep -q "\[skip-changelog\]"; then
+                      echo "✅ Changelog check bypassed via commit message"
+                      echo "Found bypass flag in one of the commit messages"
+                      echo "bypassed=true" >> $GITHUB_OUTPUT
+                  else
+                      echo "❌ CHANGELOG.md has not been modified"
+                      echo ""
+                      echo "Please update CHANGELOG.md with your changes."
+                      echo "Alternatively, include [skip-changelog] in ANY commit message to bypass this check."
+                      echo ""
+                      echo "Commit messages checked:"
+                      echo "$COMMIT_MESSAGES" | sed 's/^/  /'
+                      echo "bypassed=false" >> $GITHUB_OUTPUT
```

Comment:
It would be nice to share this text with the failure comment below to avoid duplicating it. Not sure how to do this in an actions script, but maybe a yaml reference would work?